Sources/SwiftyMonaco/MonacoViewController.swift

```diff
@@ -32,22 +32,48 @@ public class MonacoViewController: ViewController, WKUIDelegate, WKNavigationDel
         webView.layer?.backgroundColor = NSColor.clear.cgColor
         #endif
         view = webView
+        #if os(macOS)
+        DistributedNotificationCenter.default.addObserver(self, selector: #selector(interfaceModeChanged(sender:)), name: NSNotification.Name(rawValue: "AppleInterfaceThemeChangedNotification"), object: nil)
+        #endif
     }
     public override func viewDidLoad() {
         super.viewDidLoad()
         
+        loadMonaco()
+    }
+    
+    private func loadMonaco() {
         let myURL = Bundle.module.url(forResource: "index", withExtension: "html", subdirectory: "Resources")
         let myRequest = URLRequest(url: myURL!)
         webView.load(myRequest)
     }
     
+    // MARK: - Dark Mode
+    private func updateTheme() {
+        evaluateJavascript("""
+        (function(){
+            monaco.editor.setTheme('\(detectTheme())')
+        })()
+        """)
+    }
+    
+    #if os(macOS)
+    @objc private func interfaceModeChanged(sender: NSNotification) {
+        updateTheme()
+    }
+    #else
+    public override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        updateTheme()
+    }
+    #endif
```

Comment:
Handling for interface changes on macOS and iPadOS with calling for `updateTheme`